File: src/flexible-space.cpp
// INCOMPLETE

#include<bits/stdc++.h>
using namespace std;

int main(){
    int w, p; cin >> w >> p;
    vector<int> parts;
    parts.push_back(0);
    for(int i = 0; i < p; i++){
        int a; cin >> a;
        parts.push_back(a);
    }
    parts.push_back(w);
    vector<int> possibs;
    for(int i = p + 2; i >= 0; i--){
        for(int j = i - 1; j >= 0; j--){
            if(find(possibs.begin(), possibs.end(), parts[i] - parts[j]) == possibs.end()){
                possibs.push_back(parts[i] - parts[j]);
            }
        }
    }
    sort(possibs.begin(), possibs.end());
    for(int i : possibs){
        cout << i << " ";
    }
    // for(int i : parts){
    //     cout << i << " ";
    // }
    return 0;
}

File: src/flexible-space.py
w, p = map(int, input().split())
parts = list(map(int, input().split()))
parts.append(w)
parts.insert(0, 0)
possibs = []
for i in range(len(parts) - 1, -1, -1):
    for j in range(i - 1, -1, -1):
        if(parts[i] - parts[j] not in possibs):
            possibs.append(parts[i] - parts[j])
possibs.sort()
for i in possibs:
    print(i, end=" ")

# Runtime 0.05s